app/api/services/users/patch_user.py

```diff
@@ -3,15 +3,15 @@
 from api.db.models.user_models import Role, User
 from api.route.api_context import ApiContext
 from api.route.route_utils import get_or_404
+from api.route.schemas.user_schemas import UserPatchParams
 
 
 # TODO: separate controller and service concerns
 # https://github.com/navapbc/template-application-flask/issues/49#issue-1505008251
-# TODO: Use classes / objects as inputs to service methods
-# https://github.com/navapbc/template-application-flask/issues/52
-def patch_user(user_id: str, patch_data: dict, api_context: ApiContext) -> User:
+def patch_user(user_id: str, params: UserPatchParams, api_context: ApiContext) -> User:
     user = get_or_404(api_context.db_session, User, user_id)
 
+    patch_data = params.get_set_params()
```

Comment:
Re-introducing this pattern as the solution for figuring out what attributes were set by the user.